x-pack/plugins/index_lifecycle_management/__jest__/client_integration/edit_policy/serialization/policy_serialization.test.ts

```diff
@@ -137,30 +139,48 @@ describe('<EditPolicy /> serialization', () => {
       await actions.setPolicyName('test_policy');
       await actions.savePolicy();
 
-      const latestRequest = server.requests[server.requests.length - 1];
-      const entirePolicy = JSON.parse(JSON.parse(latestRequest.requestBody).body);
-
-      expect(entirePolicy).toEqual({
-        name: 'test_policy',
-        phases: {
-          hot: {
-            actions: {
-              rollover: {
-                max_age: '30d',
-                max_primary_shard_size: '50gb',
-              },
-              set_priority: {
-                priority: 100,
+      expect(httpSetup.post).toHaveBeenLastCalledWith(
+        `${API_BASE_PATH}/policies`,
+        expect.objectContaining({
+          body: JSON.stringify({
+            name: 'test_policy',
+            phases: {
+              hot: {
+                actions: {
+                  rollover: {
+                    max_age: '30d',
+                    max_primary_shard_size: '50gb',
+                  },
+                  set_priority: {
+                    priority: 100,
+                  },
+                },
+                min_age: '0ms',
               },
             },
-            min_age: '0ms',
-          },
-        },
-      });
+          }),
+        })
+      );
     });
   });
 
   describe('hot phase', () => {
+    beforeEach(async () => {
+      httpRequestsMockHelpers.setDefaultResponses();
+
+      await act(async () => {
+        testBed = await setupSerializationTestBed(httpSetup, {
+          appServicesContext: {
+            license: licensingMock.createLicense({ license: { type: 'enterprise' } }),
```

Comment:
We need to reset the state for each test run in the describe as we do with all the others and reset the license type to `enterprise` given that the last test sets it to `basic`

Reply:
nit: I don't mind having an extra `beforeEach` to reset the license type, but if `enterprise` is the default, then I would rather add an `after` block for the previous test. So the previous test would change the license to `basic` but then revert it back to the default state.

Reply:
Yeah that makes sense, I think would be better suited for this usecase!